### PR TITLE
Tensorboard Docu about Hyperparams saving

### LINKED ISSUE #1225: How to log hparams to Tensorboard?
Hello! I'm trying to view my hparams on tensorboard, but can't actually see them there. As I understood from documentation, to log hparams one should add self.hparams in the __init__ of the LightningModule. Here's what I'm doing:
```python
class MyModule(pl.LightningModule):
    def __init__(self,hparams):
        super().__init__()
        self.hparams =  Namespace(**{'learning_rate': hparams.learning_rate,
                                    'batch_size':hparams.batch_size,
                                    'normalize_embeddings':hparams.normalize_embeddings,
                                    'normalize': hparams.normalize,
                                    'k_neighbors':hparams.k_neighbors,
                                    'melspec_dir':hparams.melspec_dir})
```
My hparams also contain Trainer hparams, so to log only the right ones, I wrote this workaround. But anyway, I could not see any of my hparams. What could be the problem?

### PR DESCRIPTION
**Pull request by Skyy93**
## What does this PR do?

Adds a bit documentation to the tensorboard logger to clarify its behavior in the case of saving hyperparameters.

Fixes #1225

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/loggers/tensorboard.py [hunk 1]
```diff
@@ -146,6 +146,15 @@ def experiment(self) -> SummaryWriter:
     @rank_zero_only
     def log_hyperparams(self, params: Union[Dict[str, Any], Namespace],
                         metrics: Optional[Dict[str, Any]] = None) -> None:
```
Comment (Borda) [inline pytorch_lightning/loggers/tensorboard.py:148]:
```suggestion
    def log_hyperparams(
        self,
        params: Union[Dict[str, Any], Namespace],
        metrics: Optional[Dict[str, Any]] = None,
    ) -> None:
```

File: pytorch_lightning/loggers/tensorboard.py [hunk 2]
```diff
@@ -146,6 +146,15 @@ def experiment(self) -> SummaryWriter:
...
+        """
+        Record hyperparameters. Tensorboard logs with and without saved hyperparameters
+        are incompatible, the hyperparameters are then not displayed in the Tensorboard.
+        Please delete or move the previously saved logs to display the new ones with hyperparameters.
+
+        Args:
+            params: :class:`~argparse.Namespace` containing the hyperparameters
```
Comment (Borda) [inline pytorch_lightning/loggers/tensorboard.py:155]: mind add also `metrics`?

Comment (Skyy93) [reply pytorch_lightning/loggers/tensorboard.py:158]: sorry I overlooked it

**Review by tchaton — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

File: pytorch_lightning/loggers/tensorboard.py [hunk 3]
```diff
@@ -144,8 +144,21 @@ def experiment(self) -> SummaryWriter:
...
-                        metrics: Optional[Dict[str, Any]] = None) -> None:
+    def log_hyperparams(
+        self,
+        params: Union[Dict[str, Any], Namespace],
+        metrics: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """
+        Record hyperparameters. Tensorboard logs with and without saved hyperparameters
+        are incompatible, the hyperparameters are then not displayed in the Tensorboard.
+        Please delete or move the previously saved logs to display the new ones with hyperparameters.
```
Comment (awaelchli) [inline pytorch_lightning/loggers/tensorboard.py:155]:
```suggestion
        Record hyperparameters. TensorBoard logs with and without saved hyperparameters
        are incompatible, the hyperparameters are then not displayed in the TensorBoard.
        Please delete or move the previously saved logs to display the new ones with hyperparameters.
```

File: pytorch_lightning/loggers/tensorboard.py [hunk 4]
```diff
@@ -144,8 +144,21 @@ def experiment(self) -> SummaryWriter:
...
+
+        Args:
+            params: :class:`~argparse.Namespace` containing the hyperparameters
```
Comment (awaelchli) [inline pytorch_lightning/loggers/tensorboard.py:158]:
```suggestion
            params: a dictionary-like container with the hyperparameters
```

**Review by awaelchli — APPROVED** [review]
